### PR TITLE
Implement static namespace injection and GCS locking for integration tests

### PR DESCRIPTION
**Pull request by shubpal07**
### Objective
Resolve resource exhaustion issues (such as VPC peering limits and subnet conflicts) during concurrent integration test runs by enforcing isolated, predictable static namespaces and distributed locking.

### Problem Statement
Currently, Integration Tests generate pseudo-random hashes via `gcluster` for network and storage modules during dynamic deployments. When 60+ daily and PR tests run simultaneously, they overwhelm the central project's quotas by spinning up hundreds of isolated VPCs. Attempting to use static names natively in blueprints directly would result in concurrent tests colliding over the same exact VPC or Filestore name.

### Solution & Architecture
This PR introduces **Static Namespace Injection** paired with **GCS-Based Global Test Locks** to safely multiplex and isolate overlapping test runs without corrupting our example blueprints.

**1. GCS Distributed Locking (`gsutil cp -n`)**
We now utilize atomic GCS copy operations as a global lock server. Before any integration test is allowed to execute `gcluster create`, Ansible establishes a lock for that specific `test_name` (e.g., `gs://<project_id>-hpc-toolkit-locks/locks/daily-slurm-gke.lock`).
* If a PR test and a Daily test run the exact same blueprint, their unique `_TEST_PREFIX` ensures they deploy onto different VPCs.
* If two overlapping runs attempt to use the exact same prefix+test name, the second runner will properly wait for the lock to release, avoiding a collision.

**2. In-Flight Dynamic Settings Injection (`inject_static_names.py`)**
Rather than asking users to hardcode static names inside all of our `examples/` folder (which would break the user experience), this PR introduces a Python pre-processor to the Ansible deployer pipeline. 
Right before executing `gcluster create`, Ansible copies the target blueprint into a temporary `.injected.yaml` file. The Python script parses the topology and intelligently forces `settings.network_name`, `settings.subnetwork_name`, and `settings.name` (for Lustre/Filestore) into the deployment groups. It handles multi-NIC architecture gracefully by assigning `-net`, `-n1`, `-n2`, etc.

**3. Cloud Build Propagation**
Updated all 72 `tools/cloud-build/daily-tests/builds/*.yaml` files to explicitly forward `_TEST_PREFIX` down into the Ansible runtime environment as an extra variable, correctly differentiating `daily-` vs `pr-` namespaces.

### Verification Performed
* Ran a local highly-concurrent Python simulation across 189 simultaneous simulated tests mimicking PRs and Daily crons entirely.
* Confirmed a `100%` success rate of GCS test locks holding and deferring overlapping jobs.
* Confirmed advanced blueprint corner cases (multi-NIC topologies, multi-deployment groups, and blueprints natively lacking `settings:` blocks) are all parsed correctly by the injection script.
* Pre-commit (yamllint, python formatters) passed successfully.

### Submission Checklist

NOTE: Community submissions can take up to 2 weeks to be reviewed.

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cluster Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: tools/cloud-build/daily-tests/ansible_playbooks/scripts/inject_static_names.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                            sub_name = f"{static_test_name}-n{net_count}-sub"
+
+                        mod["settings"]["network_name"] = net_name
+                        mod["settings"]["subnetwork_name"] = sub_name
+                        net_count += 1
+                        
+                    elif src in ["modules/file-system/filestore", "modules/file-system/managed-lustre"]:
+                        if "settings" not in mod:
+                            mod["settings"] = {}
+                        mod["settings"]["name"] = f"{static_test_name}-{mod_id}"
```
[the remark this answers is not in the document]
Comment (shubpal07) [reply tools/cloud-build/daily-tests/ansible_playbooks/scripts/inject_static_names.py]: done

File: tools/cloud-build/daily-tests/ansible_playbooks/base-integration-test.yml [hunk 2]
```diff
@@ -52,7 +52,34 @@
...
+
+  - name: Acquire Test Namespace Lock
+    ansible.builtin.shell: |
+      gsutil cp -n "{{ local_lock_file }}" "{{ lock_file_path }}"
+    register: lock_acquired
+    changed_when: lock_acquired.rc == 0
+    retries: 30  # Retry for 30 minutes
+    delay: 60    # Wait 60 seconds between retries
+    until: lock_acquired.rc == 0
+    delegate_to: localhost
```
[the remark this answers is not in the document]
Comment (shubpal07) [reply tools/cloud-build/daily-tests/ansible_playbooks/base-integration-test.yml]: done
